src/variational/distributions.jl

```diff
@@ -0,0 +1,57 @@
+import Distributions: _rand!
+
+
+function jac_inv_transform(dist::Distribution, x::T where T<:Real)
+    ForwardDiff.derivative(x -> invlink(dist, x), x)
+end
+
+function jac_inv_transform(dist::Distribution, x::AbstractArray{T} where T <: Real)
+    ForwardDiff.jacobian(x -> invlink(dist, x), x)
+end
+
+function jac_inv_transform(dist::Distribution, x::TrackedArray{T} where T <: Real)
+    Tracker.jacobian(x -> invlink(dist, x), x)
+end
+
+function center_diag_gaussian(x, μ, σ)
+    # instead of creating a diagonal matrix, we just do elementwise multiplication
+    (σ .^(-1)) .* (x - μ)
```

Comment:
What are the expected types of `x` and `μ` here? Should this whole thing not be fused?
Also, assuming that everything _should_ be fused, could this please be written as the cleaner
```julia
center_diag_gaussian(x, μ, σ) = (x .- μ) ./ σ
```

Reply:
That's much nicer!

I'm wondering if there's a better abstract for  these "reparameterizations" though. Something like having a `<: Reparameterization` type which also have an `inverse` implemented.